src/tools.cc

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "public/tools.h"
+
+#include "edit_distance.h"
+#include "state.h"
+
+namespace ninja {
+
+Node* SpellcheckNode(State* state, const std::string& path) {
+  const bool kAllowReplacements = true;
+  const int kMaxValidEditDistance = 3;
+
+  int min_distance = kMaxValidEditDistance + 1;
+  Node* result = NULL;
+  for (State::Paths::iterator i = state->paths_.begin(); i != state->paths_.end(); ++i) {
```

Comment:
Seems like this function would be better as a call to std::min with a comparitor passed in (e.g. a C++11 lambda)